src/background/initial-visualization-store-data-generator.ts

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import { VisualizationConfigurationFactory } from 'common/configs/visualization-configuration-factory';
+import { EnumHelper } from 'common/enum-helper';
+import { DetailsViewPivotType } from 'common/types/store-data/details-view-pivot-type';
+import {
+    TestsEnabledState,
+    VisualizationStoreData,
+} from 'common/types/store-data/visualization-store-data';
+import { VisualizationType } from 'common/types/visualization-type';
+
+export class InitialVisualizationStoreDataGenerator {
+    constructor(
+        private readonly visualizationConfigurationFactory: VisualizationConfigurationFactory,
+    ) {}
+
+    public generateInitialState(
+        persistedData: VisualizationStoreData = null,
+    ): VisualizationStoreData {
+        const defaultTests: TestsEnabledState = {
+            adhoc: {},
+            assessments: {},
+            mediumPass: {},
+        };
+
+        if (this.visualizationConfigurationFactory != null) {
+            EnumHelper.getNumericValues(VisualizationType).forEach((test: VisualizationType) => {
+                const config = this.visualizationConfigurationFactory.getConfiguration(test);
+                defaultTests[config.testMode][config.key] = {
+                    enabled: false,
+                };
+            });
+
+            Object.keys(defaultTests.assessments).forEach(key => {
+                defaultTests.assessments[key].stepStatus = {};
+            });
+
+            Object.keys(defaultTests.mediumPass).forEach(key => {
+                defaultTests.mediumPass[key].stepStatus = {};
+            });
+        }
+        const tests: TestsEnabledState = persistedData?.tests
+            ? { ...defaultTests, ...persistedData.tests }
+            : defaultTests;
+
+        const defaultValues: VisualizationStoreData = {
+            tests,
+            scanning: null,
```

Comment:
This will throw away any persisted values for all these non-test properties, no?